plugin/supervisor.go

```diff
@@ -103,6 +112,24 @@ func (sup *supervisor) Hooks() Hooks {
 	return sup.hooks
 }
 
+// Pings the plugin through RPC
+func (sup *supervisor) Ping() error {
```

Comment:
What information or other benefit do we get from Ping that we do not get from invoking the Healthcheck hook?

Reply:
Most plugins will not have the HealthCheck hook implemented, especially at the time of initial deploy of this PR's functionality. If we can make a default implementation that does some diagnostics, or some secondary HealthCheck hook that is always our own implementation, then that can take precedence. Though, I believe the `Ping` method is useful because:
1. The method is guaranteed to be available without any further code on our part.
2. It specifically tests that the RPC connection is still alive and healthy. That is its sole purpose in the go-plugin library.
3. It's quick and cheap, other than network or timeout issues.